### PR TITLE
[controller] quote all paths incase any space char.

### PR DESCRIPTION
**Pull request by phoenix-xhuang**

### CODE REVIEW

**Comment by IJtLJZ8Rm4Yr** [comment]
> args is required for all calls and should be a string, or a sequence of program arguments. Providing a sequence of arguments is generally preferred, as it allows the module to take care of any required escaping and quoting of arguments (e.g. to permit spaces in file names). 

https://docs.python.org/3/library/subprocess.html#frequently-used-arguments

```python
subprocess.run(['ls', '-l'])
```

is better than

```python
subprocess.run('ls -l')
```

File: ymir/backend/src/ymir-controller/controller/invoker/invoker_cmd_filter.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def invoke(self) -> backend_pb2.GeneralResp:
                                                f"expected: {expected_type} vs actual: {self._request.req_type}")
 
         # invoke command
-        filter_command = "cd {0} && {1} filter --dst-rev {2} --src-revs {3} -w {4}".format(
+        filter_command = "cd \'{0}\' && {1} filter --dst-rev {2} --src-revs {3} -w \'{4}\'".format(
```
Comment (fenrir-z) [inline ymir/backend/src/ymir-controller/controller/invoker/invoker_cmd_filter.py]: 这里也改成 f-string 吧

File: ymir/backend/src/ymir-controller/controller/invoker/invoker_cmd_merge.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def invoke(self) -> backend_pb2.GeneralResp:
             return utils.make_general_response(CTLResponseCode.MIS_MATCHED_INVOKER_TYPE,
                                                f"expected: {expected_type} vs actual: {self._request.req_type}")
 
-        command = "cd {0} && {1} merge --dst-rev {2} -s {3} -w {4}".format(
+        command = "cd \'{0}\' && {1} merge --dst-rev {2} -s {3} -w \'{4}\'".format(
```
Comment (fenrir-z) [inline ymir/backend/src/ymir-controller/controller/invoker/invoker_cmd_merge.py]: 同上

**Comment by phoenix-xhuang** [comment]
Done

> > args is required for all calls and should be a string, or a sequence of program arguments. Providing a sequence of arguments is generally preferred, as it allows the module to take care of any required escaping and quoting of arguments (e.g. to permit spaces in file names).
> 
> https://docs.python.org/3/library/subprocess.html#frequently-used-arguments
> 
> ```python
> subprocess.run(['ls', '-l'])
> ```
> 
> is better than
> 
> ```python
> subprocess.run('ls -l')
> ```

File: ymir/backend/src/ymir-controller/controller/invoker/invoker_cmd_merge.py (hunk 2 above)
Comment (phoenix-xhuang) [reply ymir/backend/src/ymir-controller/controller/invoker/invoker_cmd_merge.py]: 已换过了

File: ymir/backend/src/ymir-controller/controller/invoker/invoker_cmd_filter.py (hunk 1 above)
Comment (phoenix-xhuang) [reply ymir/backend/src/ymir-controller/controller/invoker/invoker_cmd_filter.py]: 换成新的调用sh模式了，直接传List进去，不需要做转义了

**Review by under-chaos — APPROVED** [review]

**Review by IJtLJZ8Rm4Yr — APPROVED** [review]

**Pull request merged**